### PR TITLE
[EFX-526] Create basic effect contract

### PR DESCRIPTION
**Pull request by JrSchild**
### Context
Link to ticket: https://effectai.atlassian.net/browse/EFX-526

### What has been done
- See the [docker-eos PR](https://github.com/effectai/docker-eos/pull/2)

### How to test
- See the [docker-eos PR](https://github.com/effectai/docker-eos/pull/2)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,120 @@
...
+Note:
+- If you have allowance, can you also use them to lock tokens in the name of someone else?
+
+##### `token::unlock (to: name | time: uint64_t)`
+Unlock all tokens locked for `to` at given `time`.
+
+### Considerations
+Transfer by id, get the RAM back when a transfer is fulfilled.
+
+Instead of giving others an allowance, could it be worth it to create a payment request. With the following options to be groomed:
```
Comment (jeisses) [inline README.md:101]: The purpose of allowance is to have a smart contract transact on behalf of the user, e.g. for pay-as-you-go models and subscriptions. Its mainly a convenience thing so the user doesn't have to keep being reminded of signing transactions

File: README.md [hunk 2]
```diff
@@ -1,2 +1,120 @@
...
+  - Allow higher values
+  - Allow lower values
+  - Allow any value > 0
+  - When a lower/different value is sent, should the payrequest be
+    - deleted
+    - modified to be lowered
+- Hell you can make an entire debt system like this.
+- Add interest rates over time because why not.
+
+Karma implements a lock that is true by default. This locks any transactions until these are unlocked by the issuer, is this something that we want?
```
Comment (jeisses) [inline README.md:115]: I don't quite understand, could you give an example of how this works?

File: README.md [hunk 3]
```diff
@@ -1,2 +1,120 @@
...
+
+Who pays for the RAM after spending on someone elses behalf?
```
Comment (jeisses) [inline README.md:117]: The spender will have to pay and stake (like it works now)

File: README.md [hunk 4]
```diff
@@ -1,2 +1,120 @@
...
+- To remove the payment request.
+- To allow the sender to send a different value other than the one in the payment request
+  - Allow higher values
+  - Allow lower values
+  - Allow any value > 0
+  - When a lower/different value is sent, should the payrequest be
+    - deleted
+    - modified to be lowered
+- Hell you can make an entire debt system like this.
+- Add interest rates over time because why not.
```
Comment (jeisses) [inline README.md:113]: That's a cool idea! But a lending with interest system would probably fit best in a generic purpose contract

**Review by jeisses — APPROVED** [review]
It looks good. I think the lock / unlock functionality will be removed from the final contract for simplicity - it's a function that we haven't used so far. But good to have it here for reference

File: README.md (hunk 2 above)
Comment (JrSchild) [reply README.md:115]: They simply lock the contract until they specifically call an unlock action. Which is a bit strange:
https://github.com/KarmaApp/eos-karma-tokens/blob/master/eosio.token/eosio.token.cpp#L86

**Pull request merged**